lib/browser/api/desktop-capturer.ts

```diff
@@ -62,20 +64,45 @@ export async function getSources (args: Electron.SourcesOptions) {
               win.resizable = resizableValues.get(win.id);
             }
           };
-        }
+          // On Windows, the underlying WebRTC implementation does not return sources
+          // originating owned by the current process due to a Windows deadlock issue.
+          // CL: https://chromium-review.googlesource.com/c/chromium/src/+/2907415
+        } else if (process.platform === 'win32' && captureWindow) {
+          const fetches = BrowserWindow.getAllWindows().map(async (win) => {
+            let thumbnail = null;
+            if (thumbnailSize.width > 0 && thumbnailSize.height > 0) {
+              const pageContents = await win.capturePage();
```

Comment:
This captures the whole page, we only want to capture the visible viewport (which we should be able to determine and provide as x/y/width/height options to this API)

Reply:
Also worth noting that this is a thumbnail of the web-content, rather than a thumbnail of the entire window (which normally would include the window frame / controls for instance)